BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -2335,12 +2362,21 @@ void AAMGuidance()
                 }
                 else// AAM Lead
                     aamTarget = MissileGuidance.GetAirToAirTarget(TargetPosition, TargetVelocity, TargetAcceleration, vessel, out timeToImpact, optimumAirspeed);
-
-
-                if (Vector3.Angle(aamTarget - transform.position, transform.forward) > maxOffBoresight * 0.75f)
+                if(hasDataLink && (vrd != null && vrd.locked) && (Vector3.Angle(aamTarget - transform.position, transform.forward) > maxOffBoresight * 0.75f))
                 {
-                    aamTarget = TargetPosition;
+                    if(Vector3.Angle(aamTarget - transform.position, transform.forward) > maxOffBoresight){
+                        TotalDrift += DataLinkDrift;
+                        aamTarget = MissileGuidance.GetDLDeviation(aamTarget,TotalDrift);
+                        if (TargetingMode == TargetingModes.Heat) _lockTimer = 0;
+                    }
+                }
+                else if (Vector3.Angle(aamTarget - transform.position, transform.forward) > maxOffBoresight * 0.75f)
```

Comment:
This is going to mess with SARH missiles lacking a datalink; additionally, why is the missile turn rate now being clamped to boresight angle?

Reply:
The ideia here is that instead of pointing directly towards the target if it's outside the max angle(which in sum nullyfy the initial predicted position of missileguidance and also burns lots of energy depending on how hard the missile has to turn to be centered with the target) it rotate the position towards the max angle

Reply:
Also how it would mess with SARH missiles without datalink?
if it don't have datalink it enters directly the normal check that the code have before, if it has then it enters the new check